framework/src/org/checkerframework/framework/stub/StubParser.java

```diff
@@ -845,8 +880,11 @@ private void annotateParameters(
             TypeParameter param = typeParameters.get(i);
             AnnotatedTypeVariable paramType = (AnnotatedTypeVariable) typeArguments.get(i);
 
-            annotate(paramType, param.getAnnotations());
-            if (param.getTypeBound() != null && param.getTypeBound().size() == 1) {
+            if (param.getTypeBound() == null || param.getTypeBound().isEmpty()) {
+                // No bound so annotations are both lower and upper bounds
+                annotate(paramType, param.getAnnotations());
+            } else if (param.getTypeBound() != null && param.getTypeBound().size() == 1) {
+                annotate(paramType.getLowerBound(), param.getAnnotations());
                 annotate(paramType.getUpperBound(), param.getTypeBound().get(0));
             }
```

Comment:
Yes, I guess it's for code like this:

````
class A<T extends @TA Iface1 & @TB Iface2>
````

It would be good to at least issue a warning if we see > 1 bounds.